sktime/forecasting/compose/_networkpipeline.py

```diff
@@ -0,0 +1,278 @@
+# -*- coding: utf-8 -*-
+from sktime.forecasting.base._sktime import _SktimeForecaster
+from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
+from sktime.base import _HeterogenousMetaEstimator
+from sktime.utils.validation.forecasting import check_y, check_X
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+from sklearn.base import clone
+
+__author__ = ["Viktor Kazakov"]
+__all__ = ["NetworkPipelineForecaster"]
+
+
+class NetworkPipelineForecaster(
+    _OptionalForecastingHorizonMixin, _SktimeForecaster, _HeterogenousMetaEstimator
+):
+    """
+    Prototype of non sequential pipeline mimicking a network.
+
+    Each step in the pipeline can use the output of any of the previous steps or
+    the original input values to the fit and predict methods of the pipeline.
+
+    The arguments of the steps in the pipeline must be provided as a dictionary.
+    The dictionary is substituted in the fit_transform, fit or predict function calls
+    of the step transformer or estimator.
+
+    Parameters
+    ----------
+
+    steps : array of lists
+        list comprised of three elements:
+            1. name of step (string),
+            2. estimator (object),
+            3. arguments (dictionary) key value paris for
+            fit_transform or predict method of estimator
+
+    Examples
+    --------
+
+    `original_X` and `original_y` are key words referring to the
+    X and y arguments of NetworkPipelineForecaster.fit
+    and NetworkPipelineForecaster.predict
+
+    The arguments dictionary can refer to any of the previous
+    steps in the pipline by its name.
+
+    Forecasting example. Note convention for specifying
+    different arguments depending on whether
+    NetworkPipelineForecaster.fit or
+    NetworkPipelineForecaster.predict is called.
+    >>> from sktime.transformations.series.impute import Imputer
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> pipe = NetworkPipelineForecaster(steps=[
+    ...        (
+    ...            "imputer",
+    ...            Imputer(method="drift"),
+    ...            {"fit": {"Z": "original_y"}, "predict": None, "update": None},
+    ...        ),
+    ...        (
+    ...            "forecaster",
+    ...            NaiveForecaster(strategy="mean"),
+    ...            {
+    ...                "fit": {"y": "imputer", "X": "original_X", "fh": "original_fh"},
+    ...                "predict": {"fh": "original_fh"},
+    ...                "update": {"y": "original_y"},
+    ...            },
+    ...        ),
+    ...    ])
+    """
+
+    _required_parameters = ["steps"]
+    _tags = {"univariate-only": True}
+
+    def __init__(self, steps):
+        self.steps = steps
+        self._step_results = {}
+        self._fitted_estimators = {}
```

Comment:
```suggestion
        self.steps_
```
(in line with scikit-learn's trailing underscore convention for fitted variables)

Reply:
ok, will change it. I saw the `self.steps` convention being used in `TransformedTargetForecaster`.